StringPrototypeExtension.js

```diff
@@ -0,0 +1,185 @@
+String.prototype.hasVowels = function() {
+  var vowelRegex = /[aeiou]/gi;
+  var hasVowel = vowelRegex.test(this);
+  return hasVowel;
+};
+
+String.prototype.isQuestion = function() {
+  var isQuestionRegex = /\w+[?]$/;
+  return isQuestionRegex.test(this);
+};
+
+String.prototype.numberWords = function() {
+  var numberWordRegex = [
+    "zero",
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "nine"
+  ];
+  var digitsArray = this.match(/\d/g);
+  var numberToWord = "";
+  digitsArray.forEach(digit => {
+    numberToWord += numberWordRegex[digit].concat(" ");
+  });
+  return numberToWord;
+};
+
+String.prototype.wordsToArray = function() {
+  var wordRegex = /\b[0-9a-zA-Z_\-?]+\b/g;
```

Comment:
Please add a comment to complex-looking regular expression patterns like this which properly explains what it matches.